### PR TITLE
use ember-cli's debug macros

### LINKED ISSUE #235: Issue w/ ember-data
Hitting the following error when pairing with @stefanpenner:

```
Uncaught Error: Could not find module `@ember/application/deprecations` imported from `(require)`
    at missingModule (loader.js:247)
    at findModule (loader.js:258)
    at requireModule (loader.js:24)
    at eval (deprecations.js:2)
    at Object.../externals/@ember/application/deprecations.js (chunk.7a7f2b06daf33070bcbf.js:10)
    at __webpack_require__ (chunk.2b560a58216735ee49fd.js:79)
    at eval (index.js:11)
    at Module../node_modules/ember-inflector/index.js (chunk.d98733368d0da5a70d5c.js:1240)
    at __webpack_require__ (chunk.2b560a58216735ee49fd.js:79)
    at eval (relationship-meta.js:4)
```

After removing `ember-data` from package.json the error goes away.

Steps to reproduce:

1. `ember new sample-app --yarn`
2. Link embroider to sample app
3. Modify `ember-cli-build.js` with:
```
  if (process.env.EMBROIDER) {
    const Webpack = require('@embroider/webpack').Webpack;
    return require('@embroider/compat').compatBuild(app, Webpack, {
      compatAdapters: new Map([['ember-decorators', null]])
    });
  } else {
    return app.toTree();
  }
```
4. Run `EMBROIDER=true ember s`
5. Visit localhost:4200 and open up the console

### PR DESCRIPTION
**Pull request by stefanpenner**
* [fixes #235]
* Also includes https://github.com/ember-cli/babel-plugin-debug-macros/pull/79
* drop now unused deps

### CODE REVIEW

**Comment by stefanpenner** [comment]
Looks like we also have some extra dependencies, let me remove them.

**Comment by ef4** [comment]
Yes, this is what I had in mind.

File: packages/compat/src/v1-app.ts [hunk 1]
```diff
@@ -243,32 +241,6 @@ export default class V1App implements V1Package {
...
-      debugTools: {
-        isDebug,
-        source: '@ember/debug',
-        assertPredicateIndex: 1,
-      },
-    };
-    return [DebugMacros, options];
-  }
-
-  @Memoize()
```
Comment (rwjblue) [inline packages/compat/src/v1-app.ts]: Was removing `@Memoize` here intentional? This change seems unrelated to changes to the babel plugin config...

Comment (stefanpenner) [reply packages/compat/src/v1-app.ts]: good catch!

Comment (stefanpenner) [reply packages/compat/src/v1-app.ts]: fixed

**Comment by ef4** [comment]
Thanks @rwjblue I am very bad at catching that kind of thing in a code review. 😛

**Pull request merged**